### PR TITLE
Changes to Policy set API for PPRV

### PR DESCRIPTION
**Pull request by mrinalirao**
## Description

Add support to the go-tfe client for creating sentinel policy sets that will be run within the agent against a pinned policy tool version. FYI: `Pinned Policy Runtime version` is now GA.

It also removes beta tags from old OPA tests as OPA is now in GA

Api's updated:

Policy set creation (Adds agent-enabled + policy-tool-version attribute)

Testing plan
Integration tests added

External links
[JIRA](https://hashicorp.atlassian.net/browse/TF-7870)

Output from tests
Needs the following env vars:
TFE_ADDRESS
TFE_TOKEN
Set ENABLE_BETA=1

 envchain staging go test -run TestPolicySetsCreate -v ./... -tags=integration
=== RUN   TestPolicySetsCreate
=== RUN   TestPolicySetsCreate/with_valid_attributes
=== RUN   TestPolicySetsCreate/with_kind_missing
=== RUN   TestPolicySetsCreate/with_agent_enabled_missing
--- PASS: TestPolicySetsCreate (1.45s)
    --- PASS: TestPolicySetsCreate/with_valid_attributes (0.25s)
    --- PASS: TestPolicySetsCreate/with_kind_missing (0.32s)
    --- PASS: TestPolicySetsCreate/with_agent_enabled_missing (0.23s)
PASS
ok      github.com/hashicorp/go-tfe     2.317s
?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/state_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]

envchain staging go test -run TestPolicySetsUpdate -v ./... -tags=integration
=== RUN   TestPolicySetsUpdate
=== RUN   TestPolicySetsUpdate/with_valid_attributes
--- PASS: TestPolicySetsUpdate (1.61s)
    --- PASS: TestPolicySetsUpdate/with_valid_attributes (0.30s)
PASS
ok      github.com/hashicorp/go-tfe     2.301s
?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/state_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]

envchain staging go test -run TestPolicySetsList -v ./... -tags=integration
=== RUN   TestPolicySetsList
=== RUN   TestPolicySetsList/without_list_options
--- PASS: TestPolicySetsList (2.90s)
    --- PASS: TestPolicySetsList/without_list_options (0.51s)
PASS
ok      github.com/hashicorp/go-tfe     3.726s
?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/state_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]

### CODE REVIEW

**Review by nfagerlund — CHANGES_REQUESTED** [review]
I haven't fully reviewed the code yet, but there was one glaring problem (in multiple places) that jumped out at me, and I wanted to send it back to you as soon as I could. I'll take another look once you've got that sorted out!

File: policy_set.go [hunk 1]
```diff
@@ -174,14 +175,19 @@ type PolicySetCreateOptions struct {
...
 	// **Note: This field is still in BETA and subject to change.**
-	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for OPA policies.
+	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for policy evaluations.
 	Overridable *bool `jsonapi:"attr,overridable,omitempty"`
 
+	// Optional: Whether or not the policy is run as an evaluation inside the agent.
+	AgentEnabled bool `jsonapi:"attr,agent-enabled"`
+
+	// Optional: The policy tool version to run the evaluation against.
+	PolicyToolVersion string `jsonapi:"attr,policy-tool-version,omitempty"`
```
[the remark this answers is not in the document]
Comment (mrinalirao) [reply policy_set.go]: Thanks for catching that !!👍 Pushed the suggested fix. :)

File: policy_set.go [hunk 2]
```diff
@@ -224,10 +230,15 @@ type PolicySetUpdateOptions struct {
...
-	// **Note: This field is still in BETA and subject to change.**
-	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for OPA policies.
+	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for policy evaluations.
 	Overridable *bool `jsonapi:"attr,overridable,omitempty"`
 
+	// Optional: Whether or not the policy is run as an evaluation inside the agent.
+	AgentEnabled *bool `jsonapi:"attr,agent-enabled,omitempty"`
+
+	// Optional: The policy tool version to run the evaluation against.
+	PolicyToolVersion *string `jsonapi:"attr,policy-tool-version"`
```
Comment (nfagerlund) [inline policy_set.go:240]: This is still missing an `,omitempty`.

**Review by nfagerlund — CHANGES_REQUESTED** [review]
Ok, this is mostly looking good, other than one missing omitempty. My only other real concern at this point is making sure the doc comments are easy to understand. Comments inline.

File: policy_set_integration_test.go [hunk 3]
```diff
@@ -24,24 +24,53 @@ func TestPolicySetsList(t *testing.T) {
...
 	upgradeOrganizationSubscription(t, client, orgTest)
 
+	version := createAdminSentinelVersion()
+	opts := AdminSentinelVersionCreateOptions{
+		Version:          version,
+		URL:              "https://www.hashicorp.com",
+		SHA:              genSha(t),
+		Official:         Bool(false),
+		Deprecated:       Bool(false),
+		DeprecatedReason: String("Test Reason"),
```
[the remark this answers is not in the document]
Comment (mrinalirao) [reply policy_set_integration_test.go:34]: fixed 😅

File: policy_set.go [hunk 4]
```diff
@@ -174,14 +175,19 @@ type PolicySetCreateOptions struct {
 	// Optional: Whether or not the policy set is global.
 	Global *bool `jsonapi:"attr,global,omitempty"`
 
-	// **Note: This field is still in BETA and subject to change.**
 	// Optional: The underlying technology that the policy set supports
 	Kind PolicyKind `jsonapi:"attr,kind,omitempty"`
 
 	// **Note: This field is still in BETA and subject to change.**
-	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for OPA policies.
+	// Optional: Whether or not users can override this policy when it fails during a run. Only valid for policy evaluations.
```
[the remark this answers is not in the document]
Comment (mrinalirao) [reply policy_set.go:182]: yes, policy evaluations are very different from policy checks in the way they ate executed (agent vs sentinel worker), i have added a link to the docs in the comment.

**Review by nfagerlund — APPROVED** [review]

**Pull request merged**